ekncontent/ekncontent/eknc-domain.c

```diff
@@ -924,6 +924,40 @@ on_xapian_query_response (GObject *source,
     }
 }
 
+static void
+send_query_to_xapian_bridge (EkncDomain *domain,
+                             EkncXapianBridge *bridge,
+                             EkncQueryObject *query,
+                             GCancellable *cancellable,
+                             GTask *task)
+{
+  QueryState *state = g_slice_new0 (QueryState);
+  g_task_set_task_data (task, state, query_state_free);
+
+  g_autoptr(GHashTable) params = eknc_get_domain_query_params (domain);
+  eknc_xapian_bridge_query (bridge, query, params, cancellable,
+                            on_xapian_query_response, task);
+}
+
+void
+on_xapian_test_response (GObject *source,
+                         GAsyncResult *result,
+                         gpointer user_data)
+{
+  EkncXapianBridge *bridge = EKNC_XAPIAN_BRIDGE (source);
+  g_autoptr(GTask) task = user_data;
+  EkncDomain *domain = g_task_get_source_object (task);
+  GCancellable *cancellable = g_task_get_cancellable (task);
+
+  g_autoptr(GError) error = NULL;
+  eknc_xapian_bridge_test_finish (bridge, result, &error);
+  /* Ignore failures - older xapian-bridge doesn't support /test. */
```

Comment:
This will leak `error` if there is one; use `g_clear_error(&error)`.

In addition, you might want to log a message if the error is something other than the expected case where xapian-bridge is old. However, that's not a blocker for this pull request, especially since any later query will just error out if that happens.